akka-remote/src/main/scala/akka/remote/transport/Transport.scala

```diff
@@ -179,6 +179,9 @@ object AssociationHandle {
   case object Shutdown extends DisassociateInfo
   case object Quarantined extends DisassociateInfo
 
+  case class ActorSystemQuarantinedEvent(localAddress: Address, remoteAddress: Address)
+  case class RemoteShutdownEvent(localAddress: Address, remoteAddress: Address)
```

Comment:
These are user API and should live directly in `akka.remote`, plus it would be helpful to add ScalaDoc to them.